engine/api/src/org/apache/cloudstack/engine/cloud/entity/api/VirtualMachineEntity.java

```diff
@@ -115,6 +115,12 @@ String reserve(DeploymentPlanner plannerToUse, @BeanParam DeploymentPlan plan, E
     boolean stop(String caller) throws ResourceUnavailableException, CloudException;
 
     /**
+     * Stop the virtual machine, optionally force
+     *
+     */
+    boolean stop(String caller, boolean forced) throws ResourceUnavailableException, CloudException;
```

Comment:
[Flag arguments](http://martinfowler.com/bliki/FlagArgument.html) are a nasty code smell/anti-pattern.  Please consider changing method to something more descriptive with a flag argument such as ``forceStop(String caller)``.  The contrasting version of the method already exists on the interface so it would be a compatible change.